### PR TITLE
ARROW-13705: [Website] pin node version to 16 to allow CI and local dev parity

### PR DESCRIPTION
**Pull request by Jimexist**
pin node version to 16 to allow CI and local dev parity.

another reason for this pin is that e.g. node 16 uses npm v7 which has version 2 lock format, incompatible with prior versions.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -30,9 +30,11 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: ruby/setup-ruby@v1
-      - uses: actions/setup-node@v2
```
Comment (kou) [inline .github/workflows/deploy.yml:33]: Oh, I missed this in #141...

File: Rakefile [hunk 2]
```diff
@@ -23,7 +23,11 @@ webpacked_js = "javascript/main.js"
 installed_package_lock_json = "node_modules/.package-lock.json"
 
 file installed_package_lock_json => ["package.json", "package-lock.json"] do
-  sh("npm", "install", "--no-save")
+  if production?
+    sh("npm", "ci")
+  else
+    sh("npm", "install", "--no-save")
+  end
```
Comment (kou) [inline Rakefile:30]: What is the difference of `npm ci` and `npm install --no-save`?

Comment (Jimexist) [reply Rakefile:30]: https://docs.npmjs.com/cli/v7/commands/npm-ci#description

> This command is similar to npm install, except it's meant to be used in automated environments such as test platforms, continuous integration, and deployment -- or any situation where you want to make sure you're doing a clean install of your dependencies.

Comment (kou) [reply Rakefile:30]: Thanks.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**